ci-operator/step-registry/cucushift/installer/check/vsphere/custom-res-check/cucushift-installer-check-vsphere-custom-res-check-commands.sh

```diff
@@ -0,0 +1,135 @@
+   fi
+  
+   echo ${node_lists[*]}
+}
+
+function check_master_nodes_number() {
+
+#get node name
+master_lists=$(get_node_names "master")
+IFS=" "
+read -r -a master_lists <<< "${master_lists}"
+#get etcd pods names
+infra_id=$(oc get -o jsonpath='{.status.infrastructureName}' infrastructure cluster)
+mapfile -t etcd_pods_names < <(oc get pod -n openshift-etcd | grep Running | grep etcd-${infra_id} | awk '{print $1}')
+
+  if [[ ${#master_lists[@]} != "${#etcd_pods_names[@]}" && ${#etcd_pods_names[@]} != "${CONTROL_PLANE_REPLICAS}" ]]; then
+      echo "ERROR: The node number and etcd pod number are not equal to ${CONTROL_PLANE_REPLICAS}"
+      return 1
+  fi
+  count=0
+  for node in "${master_lists[@]}"; do
+      etcd="etcd-${node}"
+      if [[ ${etcd_pods_names[*]} =~ ${etcd} ]]; then
+	  count=$(($count+1))
+      fi
+  done
+  if [[ "${count}" != "${CONTROL_PLANE_REPLICAS}" ]] ; then
+     echo "ERROR: The number of masters in Ready status is ${#master_lists[@]}, etcd pod number is ${#etcd_pods_names[@]} should be ${CONTROL_PLANE_REPLICAS}" 
+     return 1
+  else 
+     return 0
+  fi  
+}
+
+function check_worker_nodes_number() {
+worker_lists=$(get_node_names "worker")
+IFS=" "
+read -r -a worker_lists <<< "${worker_lists}"
+if [[ ${#worker_lists[@]} != "${COMPUTE_NODE_REPLICAS}" ]]; then
+     echo "ERROR:The number of workers in Ready status is ${#worker_lists[@]}, should be ${COMPUTE_NODE_REPLICAS}"
+     return 1
+else
+     return 0
+fi
+
+}
+function check_node_resource() {
+    local node_type=$1 expected_disk_size=$2 expected_cpu_number=$3 expected_memory_size=$4 status=0
+    tmp=$(get_node_names $1)
+    IFS=" "
+    read -r -a node_lists <<< "${tmp}"
+    for node in "${node_lists[@]}"; do
+         disk_size=$(govc vm.info -json ${node} | jq -r ".VirtualMachines[].Config.Hardware.Device[] | select(.CapacityInKB!=null) | .CapacityInKB" | head -n1)
+         disk_size=$((${disk_size}/1024/1024))
+	 num_cpu=$(govc vm.info -json ${node} | jq -r .VirtualMachines[].Config.Hardware.NumCPU)
+	 memory_size=$(govc vm.info -json ${node} | jq -r .VirtualMachines[].Config.Hardware.MemoryMB)
+         if [[ "${disk_size}" == "${expected_disk_size}" ]]; then
+	    echo "INFO: get expected disk size!"
+	 else
+            echo "ERROR: get unexpected disk size, real disk size is ${disk_size}!"
+	    status=1
+	 fi
+         if [[ "${num_cpu}" == "${expected_cpu_number}" ]]; then
+            echo "INFO: get expected cpu number!"
+         else
+            echo "ERROR: get unexpected cpu number, real cpu number is ${num_cpu}!"
+            status=1
+         fi
+         if [[ "${memory_size}" == "${expected_memory_size}" ]]; then
+            echo "INFO: get expected memory size!"
+         else
+            echo "ERROR: get unexpected memory size, real memory size is ${memory_size}!"
+            status=1
+         fi
+    done
+    return ${status}
+}
+
+function check_machineset() {
```

Comment:
machineset spec will be passed down to vm creation in the end, if we can validate them in `check_node_resource`, I guess we  do not need to check machineset.

Reply:
so i can just delete this function and function call?

Reply:
I updated the code. now only check if worker replicas same as the number of machinesets in the function.
WDYT？

Reply:
I think we even do not check replica in machineset, the same reason, we already checked the number of master and worker nodes in `check_master_nodes_number`, we can directly drop `check_machineset` function.